src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,199 @@
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T removedValue;
+        Node<T> node = getNode(index);
+        removedValue = node.value;
+        unlink(node);
+        return removedValue;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        try {
+            Node<T> node = getNode(object);
+            unlink(node);
+            return true;
+        } catch (RuntimeException e) {
+            return false;
+        }
+    }
+
+    private void unlink(Node node) {
+        if (node.equals(head) && node.equals(tail)) {
+            head = null;
+            tail = null;
+            size--;
+        } else if (node.equals(head)) {
+            node.next.prev = null;
+            head = node.next;
+            size--;
+        } else if (node.equals(tail)) {
+            node.prev.next = null;
+            size--;
+        } else {
+            node.next.prev = node.prev;
+            node.prev.next = node.next;
+            size--;
+        }
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void ifEmptyList(Node<T> node) {
+        if (isEmpty()) {
+            head = node;
+            tail = node;
+        }
+    }
+
+    private void addIfListFilled(T value, int index) {
+        Node<T> addedNode = new Node<T>(value);
+        if (index == 0) {
+            addedNode.next = head;
+            head = addedNode;
+            addedNode.next.prev = addedNode;
+        } else {
+            Node<T> currentNode = getNode(index);
+            addedNode.prev = currentNode.prev;
+            addedNode.next = currentNode;
+            currentNode.prev.next = addedNode;
+            currentNode.prev = addedNode;
+        }
+        size++;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", size: " + size());
+        }
+    }
+
+    private void checkIndexSizeMore(int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", size: " + size());
+        }
+    }
+
+    private Node<T> getNode(int index) {
+        Node<T> current;
+        if (index <= size / 2) {
+            current = head;
+            int i = 0;
+            while (i < index) {
+                current = current.next;
+                i++;
+            }
+        } else {
+            current = tail;
+            int i = size - 1;
+            while (i > index) {
+                current = current.prev;
+                i--;
+            }
+        }
+        return current;
+    }
+
+    private Node<T> getNode(T object) {
+        Node<T> current;
+        int index = 0;
+        if (index <= size / 2) {
```

Comment:
You just created index = 0 and then making `if` block.
There is no sense. Index is always 0
When we search by object we cannot determine if we should start from head or tail. So just choose one option and search for element.

Approach is correct when we search by index and can understand it's closer to head or tail